modules/bootstrap/main.tf

```diff
@@ -1,55 +1,34 @@
-locals {
-  bootstrap_filenames = { for f in try(fileset(var.bootstrap_files_dir, "**"), {}) : f => "${var.bootstrap_files_dir}/${f}" }
-  # invert var.files map 
-  inverted_files     = { for k, v in var.files : v => k }
-  inverted_filenames = merge(local.bootstrap_filenames, local.inverted_files)
-  # invert local.filenames map
-  filenames = { for k, v in local.inverted_filenames : v => k }
-}
-
+# https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/storage_account
 resource "azurerm_storage_account" "this" {
   count = var.create_storage_account ? 1 : 0
 
   name                     = var.name
   location                 = var.location
   resource_group_name      = var.resource_group_name
-  min_tls_version          = var.min_tls_version
+  min_tls_version          = var.storage_network_security.min_tls_version
   account_replication_type = "LRS"
   account_tier             = "Standard"
+  account_kind             = "StorageV2"
```

Comment:
What do you think about defining new variable for `account_kind` with default value `StorageV2` ?

Reply:
possible, but most of the settings other than `StorageV2` requier `Premium` tier, so we would have to parametrize both